### PR TITLE
Encourage spell checking while submit PR

### PR DESCRIPTION
**Pull request by charrywanganthony**
Encourage spell checking while submit PR
ref: #7651

### CODE REVIEW

**Comment by bradtopol** [comment]
Would like @steveperry-53 input on whether this is needed in the style guide.

File: docs/home/contribute/style-guide.md [hunk 1]
```diff
@@ -399,6 +399,18 @@ considered new in a few months.
...
+## Spell Checking
+
+Checking spelling before you propose changes to documents.  
+Do check it yourself as carefully as possible. Because false positives occur at a pretty high rate by automated tools.
+
+You can check typos with the help of [misspell](https://github.com/client9/misspell#install).
+
+Misspell has been used in Kubernetes and Community project:
+
+    ./hack/verify-spelling.sh
```
Comment (heckj) [inline docs/home/contribute/style-guide.md]: I love this as an example of how to use the tool, with the exception that this repository doesn’t have a `hack` directory. Could we replace this example of how it’s used in Kubernetes with an example that you think is effective for use in writing/validating documentation in this repo?

**Comment by steveperry-53** [comment]
> Would like @steveperry-53 input on whether this is needed in the style guide.

I don't think the style guide is the right place for this. Instead, I think this should go in these two topics:

* [Creating a Documentation Pull Request](https://kubernetes.io/docs/home/contribute/create-pull-request/)

* [Participating in SIG Docs](https://kubernetes.io/docs/home/contribute/participating/)

And possibly in this topic: [Writing a New Topic](https://kubernetes.io/docs/home/contribute/write-new-topic/)

**Comment by steveperry-53** [comment]
I don't think we have to point out that Misspell is used in other repos. I would suggest something like this:

It is a good idea to check spelling both manually and with a tool. You can use the Misspell tool by entering this command:

    <some-command>

**Comment by zacharysarah** [comment]
@steveperry-53 I like your replacement suggestion. Adding a link to Misspell would also be helpful.

File: docs/home/contribute/style-guide.md (hunk 1 above)
Comment (zacharysarah) [reply docs/home/contribute/style-guide.md]: Please implement @steveperry-53's [feedback](https://github.com/kubernetes/website/pull/7691#issuecomment-371884158) here.

**Comment by charrywanganthony** [comment]
@zacharysarah @heckj @steveperry-53 @bradtopol Modified. PTAL. :smile:

**Comment by charrywanganthony** [comment]
Before this PR merges, typos in #7710 should be fixed.
And there's still a problem:
`docs/user-journeys/users/cluster-operator/_advanced.md:76:457: "ridiculus" is a misspelling of "ridiculous"` what's this file?

File: docs/home/contribute/create-pull-request.md [hunk 2]
```diff
@@ -44,12 +44,30 @@ choose the
 [page type](/docs/home/contribute/page-templates/)
 that is the best fit for your content.
 
+## Spell Checking
+
+Check the spelling of any proposed changes. Verify any changes made by an automated spellchecker.
+
+You can check typos with the help of [Misspell](https://github.com/client9/misspell) which has been used in Kubernetes and Community project.
```
Comment (steveperry-53) [inline docs/home/contribute/create-pull-request.md]: comma after misspell)

File: docs/home/contribute/create-pull-request.md [hunk 3]
```diff
@@ -44,12 +44,30 @@ choose the
...
+
+1. If you use GO, you can install the Misspell tool by entering this command:
```
Comment (steveperry-53) [inline docs/home/contribute/create-pull-request.md]: GO -> Go

Comment (charrywanganthony) [reply docs/home/contribute/create-pull-request.md]: @steveperry-53 Modified.

File: docs/home/contribute/create-pull-request.md [hunk 4]
```diff
@@ -44,12 +44,30 @@ choose the
...
+Check the spelling of any proposed changes. Verify any changes made by an automated spellchecker.
+
+You can check typos with the help of [Misspell](https://github.com/client9/misspell), which has been used in Kubernetes and Community project.
+
+1. If you use Go, you can install the Misspell tool by entering this command:
+    ```
+    go get -u github.com/client9/misspell/cmd/misspell
+    ```
+   Or you can install the tool [in other ways](https://github.com/client9/misspell#install).
+
```
Comment (steveperry-53) [inline docs/home/contribute/create-pull-request.md]: Go getting misspell is not enough to run it. You have to build it first. So I think we need to show that as a step.

1. Get misspell.
2. Build misspell.
3. Run misspell.

And we need to mention that misspell has to be on your path. Either that or you have to specify the full path to the misspell command: ... | xargs ~/bin/missspell

File: docs/home/contribute/create-pull-request.md [hunk 5]
```diff
@@ -44,12 +44,30 @@ choose the
...
+2. Then you can run this command under the root directory of your kubernetes/website repository:  
+    ```
+    git ls-files | grep -v generated |grep -v vendor | grep -v imported | grep -v api-reference | xargs misspell
+    ```
```
Comment (steveperry-53) [inline docs/home/contribute/create-pull-request.md]: I'm having trouble getting the expected results from this tool. I intentionally put several misspelled words in docs/tasks/run-application/delete-stateful-set.md.

"You can delet a StatefulSet in the samee way you delate oter ridiculus resorces in Kubernetes: use the `kubectl delete` command, and spesify the StatefulSet either by fiele or by name."

And yet the only thing that misspell detects is "ridiculus".

```
git ls-files | grep -v generated |grep -v vendor | grep -v imported | grep -v api-reference | xargs ~/bin/misspell
docs/tasks/run-application/delete-stateful-set.md:27:61: "ridiculus" is a misspelling of "ridiculous"
docs/user-journeys/users/cluster-operator/_advanced.md:13:278: "ridiculus" is a misspelling of "ridiculous"
```

**Comment by steveperry-53** [comment]
_advanced.md is an unpublished file that has a bunch of latin placeholder text.

File: docs/home/contribute/create-pull-request.md (hunk 5 above)
Comment (CaoShuFeng) [reply docs/home/contribute/create-pull-request.md]: This tool use a word list to detect misspelling:
https://github.com/client9/misspell#where-do-the-word-lists-come-from

File: docs/home/contribute/create-pull-request.md (hunk 4 above)
Comment (charrywanganthony) [reply docs/home/contribute/create-pull-request.md]: `go get` command should do install itself completely. But maybe we can't list all the solutions in this document(like all the chinese contributor should set proxy...). So i add a short description and a link.

File: docs/home/contribute/create-pull-request.md (hunk 5 above)
Comment (charrywanganthony) [reply docs/home/contribute/create-pull-request.md]: `delet` and `samee` seems not in the dictionary. 
I set the command ignore the file which name contains `_advanced`. If you have any other file or folder that we should not check, plz let me know.

File: docs/home/contribute/create-pull-request.md (hunk 4 above)
Comment (steveperry-53) [reply docs/home/contribute/create-pull-request.md]: Thanks, I didn't realize that the go get command did an installation.

**Comment by Bradamant3** [comment]
Like @steveperry-53, I'm concerned about the emphasis on a single tool that appears designed not for standard spellchecking but for spellchecking in code files. There are other spellchecking tools that do the job, better in my experience. Instead of emphasizing a single tool that must be built, I suggest that we do further research, identify a few tools that we can recommend, and point to the tool documentation as the place to get started.
The tools I'm aware of can also be used as more extensive doc linters, although they don't need to be implemented that way.
There are also spellcheck plugins for Visual Studio Code, which most of the K8s engineers I know use, for Atom, and for Sublime. I'm sure there are more out there for other editors. These plugins tend to rely on the same dictionaries from what I can tell (Aspell or Hunspell) -- dictionaries that are far more substantial than misspell's. From the misspell docs, it appears that the author came up with their own word list. This isn't adequate for a docs repository, nor, I suspect for any repo that contains a significant amount of English prose.
Strongly recommend closing this PR in favor of an issue and further investigation and discussion in sig-docs.

**Comment by Bradamant3** [comment]
One more thought, following up on a suggestion I made in the original misspell PR to this repo:
I do think that *eventually* it would be useful/good/important to customize a linter and integrate it into the build. But this is a project that requires careful thought and planning, and shouldn't be implemented until after we've made more progress developing a docs culture around paying better attention to the style guide in initial commits (that is, locally), and around solid local spellchecking. If contributors don't understand what the linter/spellcheck integration is for and how it works, it becomes a blocker, not a tool for improving quality.

**Pull request closed without merging**